### ISSUE TITLE
DeepImpact / uniCOIL is much slower than BM25?

### ISSUE THREAD

**Issue by jingtaozhan** [castorini/pyserini#723]
I haven't been able to reproduce DeepImpact and uniCOIL because the current release does not support impact retrieval. But I am curious about why the doc says they are much slower than BM25.

> Query evaluation is much slower than with bag-of-words BM25; a complete run can take around half an hour. Note that the important option here is -impact, where we specify impact scoring.

**Comment by MXueguang** [comment]
Hi @jingtaozhan,

If you are interested in reproducing DI/uniCOIL right now, you have to follow the [development installation](https://github.com/castorini/pyserini#development-installation)
Regarding the speed, we had some discussion here https://github.com/castorini/anserini/pull/1600.

TL;DR 1. uniCOIL has a smaller vocabulary size. 2. uniCOIL didn't remove stopwords, punctuations.
Filter out terms with idf<1 is an easy way to speed up searching.

**Comment by jingtaozhan** [comment]
I see. Thank you!
